game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/RemoveUnprotectedUnits.java

```diff
@@ -0,0 +1,173 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_UNESCORTED_TRANSPORTS;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.AllArgsConstructor;
+import org.triplea.java.collections.CollectionUtils;
+
+@AllArgsConstructor
+public class RemoveUnprotectedUnits implements BattleStep {
+
+  private static final long serialVersionUID = 4357860848979564096L;
+
+  protected final BattleState battleState;
+
+  protected final BattleActions battleActions;
+
+  @Override
+  public List<String> getNames() {
+    final List<String> steps = new ArrayList<>();
+    if (battleState.getBattleSite().isWater()
```

Comment:
Nit: Have you considered using a ternary here? Avoiding intermediate variables is often a good thing.

Reply:
I don't prefer ternary operations when the condition is multiple lines.  I'll get rid of the intermediate variables by just returning a new List in each return.

Reply:
Fair enough, something to consider though. Having a method lead with `return` is  a powerful clue to a reader. Otherwise with an 'if' and 'otherwise' block here, it's arguably cognitively more complex. Specifically until you've scanned the whole method do you realize that all it is is a condition and a return statement based on that (and the simplest form of that is a ternary which leads with the keyword 'return' to help tell a reader that information immediately)